### ISSUE TITLE
Multi-line strings syntax highlighting

### ISSUE THREAD

**Issue by Leakbang** [pfgithub/language-zig-zls#4]
There is a tiny issue with the syntax highlighting of multi-line strings. For example in the code below the 'comment highlighting' (default: green) should stop at the commented line.

```zig
fn function() void {
    const text = std.fmt.comptimePrint(
    \\ Multi
    \\ Line 
    \\ Strings
    \\ {variable}
    // Comment coloring/Highlighting should stop here <-
    , .{.variable = "Are cool"});
    std.log.default.info("{s}", .{text});
}
```
But this is what happens:
![image](https://user-images.githubusercontent.com/51197897/167310697-27e4ed8f-dd16-4d0c-b793-bcb6ec242981.png)

The whole rest of the code becomes green :green_book:

**Comment by Skehmatics** [comment]
Does not happen for me (as of ZLS `0.12.0-dev.512+9ba474` and pulsar v1.116, the tech equivalent of a lifetime later). This may have been a ZLS bug

**Comment by pfgithub** [comment]
I don't think it can be a zls bug, zls doesn't provide coloring. It's likely an issue with the tree-sitter syntax and it may be fixed this version of the grammar: https://github.com/maxxnino/tree-sitter-zig/blob/main/src/grammar.json .

Since the existing grammar was textmate, it's likely that an atom update fixed it. That was an issue I've seen with textmate grammars before that happened on atom but not vscode.

**Comment by RedCMD** [comment]
works fine for me
using the grammar in VSCode

<img width="668" height="292" alt="Image" src="https://github.com/user-attachments/assets/5c6a12c5-fc16-4bad-9e1f-94da86a5d943" />

here's my VSCode extension to help with authoring TextMate grammars https://marketplace.visualstudio.com/items?itemName=RedCMD.tmlanguage-syntax-highlighter
